tests/test_mailchimp.py

```diff
@@ -18,69 +20,60 @@ def setUp(self, *args, **kwargs):
         with self.context:
             handlers.celery_before_request()
 
-    @mock.patch('website.mailchimp_utils.get_mailchimp_api')
-    def test_get_list_id_from_name(self, mock_get_mailchimp_api):
-        list_name = 'foo'
-        mock_client = mock.MagicMock()
-        mock_get_mailchimp_api.return_value = mock_client
-        mock_client.lists.list.return_value = {'data': [{'id': 1, 'list_name': list_name}]}
-        list_id = mailchimp_utils.get_list_id_from_name(list_name)
-        mock_client.lists.list.assert_called_with(filters={'list_name': list_name})
-        assert_equal(list_id, 1)
-
     @mock.patch('website.mailchimp_utils.get_mailchimp_api')
     def test_get_list_name_from_id(self, mock_get_mailchimp_api):
         list_id = '12345'
         mock_client = mock.MagicMock()
         mock_get_mailchimp_api.return_value = mock_client
-        mock_client.lists.list.return_value = {'data': [{'id': list_id, 'name': 'foo'}]}
+        mock_client.lists.get.return_value = {'id': list_id, 'name': 'foo'}
         list_name = mailchimp_utils.get_list_name_from_id(list_id)
-        mock_client.lists.list.assert_called_with(filters={'list_id': list_id})
+        mock_client.lists.get.assert_called_with(list_id=list_id)
         assert_equal(list_name, 'foo')
 
     @mock.patch('website.mailchimp_utils.get_mailchimp_api')
     def test_subscribe_called_with_correct_arguments(self, mock_get_mailchimp_api):
-        list_name = 'foo'
+        list_name = MAILCHIMP_GENERAL_LIST
         user = UserFactory()
         mock_client = mock.MagicMock()
         mock_get_mailchimp_api.return_value = mock_client
-        mock_client.lists.list.return_value = {'data': [{'id': 1, 'list_name': list_name}]}
+        mock_client.lists.get.return_value = {'id': 1, 'list_name': list_name}
         list_id = mailchimp_utils.get_list_id_from_name(list_name)
         mailchimp_utils.subscribe_mailchimp(list_name, user._id)
         handlers.celery_teardown_request()
-        mock_client.lists.subscribe.assert_called_with(
-            id=list_id,
-            email={'email': user.username},
-            merge_vars={
-                'fname': user.given_name,
-                'lname': user.family_name,
-            },
-            double_optin=False,
-            update_existing=True,
+        mock_client.lists.members.create.assert_called_with(
+            list_id=list_id,
+            data={
+                'status': 'subscribed',
+                'email_address': user.username,
+                'merge_fields': {
+                    'FNAME': user.given_name,
+                    'LNAME': user.family_name
+                }
+            }
```

Comment:
nit: I don't love `assert_called_with`, since it's crazy brittle (as demonstrated here). Is just `assert_called` sufficient? Or else assert a subset of the especially relevant kwargs (which is still brittle if the signature completely breaks, but at least requires less to change)?